perceval/backend.py

```diff
@@ -106,11 +106,45 @@ def archive(self, obj):
 
         self._archive = obj
 
-    def fetch(self, from_date=DEFAULT_DATETIME):
+    def fetch_items(self, **kwargs):
         raise NotImplementedError
 
-    def fetch_from_cache(self):
-        raise NotImplementedError
+    def fetch(self, category, **kwargs):
```

Comment:
Both `fetch` and `fetch_from_archive` must iterate over the items generated by `fetch_items`. Each item should be 
"decorated" with infomation from `metadata` function. It should be something like:

```
def fetch(self, category, **kwargs):
    if self.archive:
        self.archive.init_metadata(self.origin, self.__name__, self.version, category, kwargs)

    self.client = self._init_client()

    for item in self.fetch_items(**kwargs):
        yield metadata(item)
```

In fact, `fetch` and `fetch_from_archive` can use again a generic method with the code from above. What the code will change is the way archive is initialized and used.